add better errors for our calls to Setsockopt()

squeed opened the pull request:
Some users reported this failing, but we don't have enough logging to figure out what, exactly, is failing.

pippolo84 commented on pkg/fqdn/dnsproxy/udp.go:
```diff
@@ -59,42 +59,49 @@ var rawconn6 *net.IPConn // raw socket for sending IPv6
-				if opErr == nil && !option.Config.EnableBPFTProxy {
-					opErr = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEPORT, 1)
+				if err := unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEADDR, 1); err != nil {
+					opErr = fmt.Errorf("setsockopt(SO_REUSEADDR) failed: %w", err)
```
If `opErr != nil` we should also add a `return` in order to avoid calling `setsockopt(SO_REUSEPORT)` in the if branch `!option.Config.EnableBFTProxy` below.

squeed replied:
good catch, yup.

pippolo84 commented on pkg/fqdn/dnsproxy/udp.go:
```diff
@@ -59,42 +59,49 @@ var rawconn6 *net.IPConn // raw socket for sending IPv6
+				}
+				if !option.Config.EnableBPFTProxy {
+					if err := unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEPORT, 1); err != nil {
+						opErr = fmt.Errorf("setsockopt(SO_REUSEPORT) failed: %w", err)
```
Since this is the last `setsockopt` it does not matter, but I'd add a return here too in case of error, to prevent overwriting it accidentally if someone adds a further setsockopt call. But maybe I'm being overly cautious, I'll leave it up to you.

squeed replied:
symmetry is good :-)

christarazi commented on daemon/cmd/sockopt.go:
```diff
@@ -4,28 +4,31 @@
-		soerr = unix.SetsockoptInt(s, unix.SOL_SOCKET, unix.SO_REUSEPORT, 1)
+		if err := unix.SetsockoptInt(s, unix.SOL_SOCKET, unix.SO_REUSEPORT, 1); err != nil {
+			soerr = fmt.Errorf("failed to Setsockopt(SO_REUSEPORT): %w", err)
+		}
```
Nit: In one error msg, `setsockopt` is lower-cased and the other is upper-cased.

christarazi reviewed (approved):
One minor nit and pending Fabio's suggestions (I don't see the changes applied), LGTM

squeed committed:
add better errors for our calls to Setsockopt()

Some users reported this failing, but we don't have enough logging to
figure out what, exactly, is failing.

Signed-off-by: Casey Callendrello <cdc@isovalent.com>

pippolo84 reviewed (approved):
LGTM :rocket:
